internal/elasticsearch/client.go

```diff
@@ -86,40 +78,34 @@ type Client interface {
 	ListTemplates() (sets.String, error)
 	GetIndexTemplates() (map[string]estypes.GetIndexTemplate, error)
 	UpdateTemplatePrimaryShards(shardCount int32) error
-
-	SetSendRequestFn(fn FnEsSendRequest)
 }
 
-type FnEsSendRequest func(cluster, namespace string, payload *EsRequest, client k8sclient.Client)
-
 type esClient struct {
-	cluster         string
-	namespace       string
-	k8sClient       k8sclient.Client
-	fnSendEsRequest FnEsSendRequest
+	cluster   string
+	namespace string
+	eoclient  elasticsearch6.Client
+	k8client  k8sclient.Client
 }
 
-type EsRequest struct {
-	Method          string // use net/http constants https://golang.org/pkg/net/http/#pkg-constants
-	URI             string
-	RequestBody     string
-	StatusCode      int
-	RawResponseBody string
-	ResponseBody    map[string]interface{}
-	Error           error
-}
-
-func NewClient(cluster, namespace string, client k8sclient.Client) Client {
+// NewClient Getting new client
+func NewClient(cluster, namespace string, client elasticsearch6.Client) Client {
 	return &esClient{
-		cluster:         cluster,
-		namespace:       namespace,
-		k8sClient:       client,
-		fnSendEsRequest: sendEsRequest,
+		cluster:   cluster,
+		namespace: namespace,
+		eoclient:  client,
 	}
 }
 
-func (ec *esClient) SetSendRequestFn(fn FnEsSendRequest) {
-	ec.fnSendEsRequest = fn
+// NewClient Getting new client
+func NewK8Client(cluster, namespace string, client k8sclient.Client) Client {
```

Comment:
Why do we have two `New*Client` functions? shouldn't we just have `NewClient` as we had before? it should return an esClient

Reply:
I just kept for testing to be finalized. Now its done so now I have just one NewClient method.